app/src/marketplace/containers/EditProductPage/index.jsx

```diff
@@ -107,19 +109,41 @@ const EditProductPage = ({ product }: { product: Product }) => {
     const originalProduct = useProduct()
     const { beneficiaryAddress } = originalProduct
 
-    useEffect(() => {
-        loadEthIdentities()
-        loadDataUnion(beneficiaryAddress)
-        loadDataUnionStats(beneficiaryAddress)
-    }, [beneficiaryAddress, loadDataUnion, loadDataUnionStats, loadEthIdentities])
-
     const isLoading = savePending || publishDialogLoading
     const modalsOpen = !!(isDataUnionDeployDialogOpen || isConfirmSaveDialogOpen || isPublishDialogOpen)
     const isDisabled = isLoading || modalsOpen
     const isDataUnion = isDataUnionProduct(product)
     // TODO: should really check for the contract existance here
     const isDeployed = isDataUnion && isEthereumAddress(product.beneficiaryAddress)
 
+    useEffect(() => {
+        if (isDataUnion) {
+            loadEthIdentities()
+        }
+    }, [
+        isDataUnion,
+        loadEthIdentities,
+    ])
+
+    useEffect(() => {
+        if (isDeployed) {
+            loadDataUnion(beneficiaryAddress)
+            loadDataUnionStats(beneficiaryAddress)
+            loadDataUnionSecrets(beneficiaryAddress)
```

Comment:
TODO: Use same stats loading from `useDataUnionServerStats()` hook and remove this controller method.